core/cap-0051.md

```diff
@@ -89,22 +89,30 @@ func $map_put (param $obj_map i64) (param $key i64) (param $val i64) (result i64
 /// Given a key, return its value from an existing map. Trap if key not found.
 func $map_get (param $obj_map i64) (param $key i64) (result i64)
 
-/// Check if a key exists in a map. Returns a static true/false value (Tag 2).
-func $map_has (param $obj_map i64) (param $key i64) (result i64)
-
 /// Given a key, delete its value from an existing map. Trap if key not found.
 func $map_del (param $obj_map i64) (param $key i64) (result i64)
 
 /// Return length of an existing map.
 func $map_len (param $obj_map i64) (result i64)
 
+/// Check if a key exists in a map. Returns a static true/false value (Tag 2).
+func $map_has (param $obj_map i64) (param $key i64) (result i64)
+
 /// Given a key, find the first element less than itself in the map's sorted order. 
 /// If such an element exists, return its key, otherwise return an SCStatus containing the error code (TBD).
-func $map_lower_bound (param $obj_map i64) (param $key i64) (result i64)
+func $map_prev_key (param $obj_map i64) (param $key i64) (result i64)
 
 /// Given a key, find the first element greater than itself in the map's sorted order. 
 /// If such an element exists, return its key, otherwise return an SCStatus containing the error code (TBD).
-func $map_upper_bound (param $obj_map i64) (param $key i64) (result i64)
+func $map_next_key (param $obj_map i64) (param $key i64) (result i64)
+
+/// Given a key, find the minimum element in the map. 
```

Comment:
There is no key given as a parameter, might I suggest:
```suggestion
/// Map min key returns the minimum key in the map. 
```

Reply:
Ah [Facepalm], my bad. Updated.